Update API Docs

devdot opened the pull request:
I came across a few minor issues in the API documentation. 

(1) In the example response for read shelf/book, the tags have `id` - but they do not have `id` just like at any other endpoint.

(2) In the example response for read book, the page within `contents` does not have a `type` - but a quick test on the demo shows that any object within `contents` has a `type`.

I also noticed some fields are missing in the example responses, e.g. pages have an `editor` field (and sometimes `url`) and users within entities also a `slug` besides `name` and `id`. I assume it's an unnecessary bloat of the example responses to add those?

devdot committed:
API Docs: Remove id from Tag in Response

devdot committed:
API Docs: Add Missing Type in Response

Type is always returned when pages/chapters are in a contents array.

ssddanbrown commented:
Thanks @devdot!

> I also noticed some fields are missing in the example responses, e.g. pages have an editor field (and sometimes url) and users within entities also a slug besides name and id. I assume it's an unnecessary bloat of the example responses to add those?

If it's just singular odd fields missing that are currently being provided in API responses, then it's probably not due to bloat, but instead there have been more recent changes, after responses were recorded, what have tangentially affected those old endpoints. Ideally those responses would correctly reflect all fields.

devdot committed:
API Docs: Add User Slugs to Example Responses

devdot committed:
API Docs: Remove Dates from Tags in Example Responses

devdot committed:
API Docs: Sort a few example responses

devdot committed:
API Docs: Add book_slug to Example Responses

Remove the book attribute in responses because it is never returned by the API. Currently, Chapters Create does not return book_slug! (The example response is consistent with the inconsistent API behavior)

devdot committed:
API Docs: Add Missing editor fields in Example Responses

devdot committed:
API Docs: Add Missing Fields in Example Responses

devdot commented:
I've gone through the example responses and made them match the actual responses as I worked through them for a project I'm working on.

One potential issue I found is that chapters create does not return `book_slug`, even though all other chapters endpoints do return that value. Adding that should be trivial, but it's not in the scope of this PR.

ssddanbrown commented:
Thank you very much for these updates @devdot!
